src/Symfony/Component/Validator/Constraints/NotSuspiciousValidator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Contracts\Translation\LocaleAwareInterface;
+
+/**
+ * @author Mathieu Lechat <mathieu.lechat@les-tilleuls.coop>
+ */
+class NotSuspiciousValidator extends ConstraintValidator implements LocaleAwareInterface
+{
+    private string $locale;
+
+    public function validate(mixed $value, Constraint $constraint)
+    {
+        if (!$constraint instanceof NotSuspicious) {
+            throw new UnexpectedTypeException($constraint, NotSuspicious::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!\is_scalar($value) && !$value instanceof \Stringable) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        $value = (string) $value;
+        if ('' === $value) {
+            return;
+        }
+
+        $checker = new \Spoofchecker();
+
+        $checker->setRestrictionLevel($constraint->restrictionLevel);
```

Comment:
This method only exists when ext-intl is compiled against ICU 58+. So we might need to make it conditional

Reply:
Indeed. Not sure if there needs to be an `else`: I didn’t understand how this was supposed to work before ICU 58.

Reply:
what's the behavior before ICU58? can't we throw an exception when ppl set this to something we cannot configure?

Reply:
As ICU’s `RESTRICTION_LEVEL` and `SINGLE_SCRIPT` have the same value I think you could only emulate `SINGLE_SCRIPT_RESTRICTIVE` before ICU58.

In that case we could throw an exception if `$restrictionLevel` is anything else?

Reply:
:100:

Reply:
[Done](https://github.com/symfony/symfony/pull/49300/commits/b48a2dd9730c7269eb5cb49eef9f7e9c4d149654). If someone could check [my assumption](https://github.com/symfony/symfony/pull/49300#discussion_r1102790698) that would be great :sweat_smile: